ACMILabs/lens-reader
# #47 Add ability to set hostname to AV Code using the Balena API

Resolves #47

Set the Lens Reader hostname using the Balena API at boot.

### Acceptance Criteria
- [x] Set hostname using the Balena API

### Relevant design files
* None

### Testing instructions
1. Run tests to check they all pass: `cd development` and `docker-compose up --build` and `docker exec -it lensreader make linttest`
1. Add your lens reader to the app: [s__tap-reader-pi-4](https://dashboard.balena-cloud.com/apps/1520826/devices)
1. Open Terminal in Balena, and run this command to check the current hostname: `curl "$BALENA_SUPERVISOR_ADDRESS/v1/device/host-config?apikey=$BALENA_SUPERVISOR_API_KEY"`
1. Add a Device Variable `SET_HOSTNAME` and set it to `true`
1. Open Terminal in Balena, and run this command to check that the hostname is now set to the device name: `curl "$BALENA_SUPERVISOR_ADDRESS/v1/device/host-config?apikey=$BALENA_SUPERVISOR_API_KEY"`
1. Check that a tap still shows up on XOS: https://xos.acmi.net.au/admin/exhibitions/tap/

### DoD
For requester to complete:
